main.go

```diff
@@ -122,50 +120,8 @@ func main() {
 	cfg.LoadEnvForStruct(opts)
 	options.Resolve(opts, flagSet, cfg)
 
-	err := opts.Validate()
```

Comment:
I think this validation makes sense in the main module.

Reply:
The reason we moved it in was because the err was being reused in the expressions that were moved into `CreateSecureProxy`.

Reply:
I see, but it looks like `CreateSecureProxy` does a lot of things--maybe more than you would think based on the name. I think option validation can continue to happen outside that function. Also, I see some calls to `os.Exit` and `log.Fatalf` in that function, which I'm used to seeing in `main` but not in other functions that also return an error.